### PR TITLE
Clean up ParcelableProxyMessage.

### PR DESCRIPTION
**Pull request by jasonwyatt**
This PR makes ParcelableProxyMessage behave more like it should - a type hierarchy, and allows us to only have one `send`/`on` method in ADL for passing ProxyMessages.

### CODE REVIEW

File: shells/android/java/arcs/storage/service/IResultCallback.aidl [hunk 1]
```diff
@@ -16,9 +16,9 @@ import arcs.crdt.parcelables.ParcelableCrdtException;
 /** Mechanism allowing an asynchronous response from the StorageService. */
 interface IResultCallback {
     /**
-     * Handles success/failure of an async op.<br><br>
+     * Called to signal a successful/erroneous result from the StorageService or ServiceStore.
      *
-     * {@param exception} is nullable.
+     * <p>{@code exception} will be {@code null} when the result indicates success.
```
Comment (csilvestrini) [inline shells/android/java/arcs/storage/service/IResultCallback.aidl]: I know you're planning to come back to fix these up properly later, but it's still bothering me.

```
@param exception will be {@code null} when the ...
```

File: src_kt_runtests.sh [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
 
-echo "Building src_kt"
+echo "Building src_kt and shells/android"
 bazel build \
   --noshow_progress \
   --noshow_loading_progress \
   --test_output=errors \
   --incompatible_depset_is_not_iterable=false \
-  //src_kt/java/... //src_kt/javatests/...
+  //src_kt/java/... //src_kt/javatests/... //shells/android/java/... //shells/android/javatests/...
```
Comment (csilvestrini) [inline src_kt_runtests.sh]: Does `//shells/android/...` work?

File: shells/android/java/arcs/storage/service/IResultCallback.aidl (hunk 1 above)
Comment (jasonwyatt) [reply shells/android/java/arcs/storage/service/IResultCallback.aidl]: Ah, ok.. I'll use this approach.

File: src_kt_runtests.sh (hunk 2 above)
Comment (jasonwyatt) [reply src_kt_runtests.sh]: Yep!

**Pull request merged**